x/staking/genesis.go

```diff
@@ -34,8 +34,11 @@ func InitGenesis(
 	keeper.SetParams(ctx, data.Params)
 	keeper.SetLastTotalPower(ctx, data.LastTotalPower)
 
+	valCache := make(map[string]types.Validator, len(data.Validators))
```

Comment:
@ValarDragon, mind taking a quick look at the modified `ImportGenesis` changes please?

Reply:
It looks good to me. One optimization that would be good though is to make the map key the direct `GetValidatorAddr()` representation, not the `GetValidatorAddr().String()` form, since that requires many bech32 encodes. (Granted these bech32 encodes may not be so costly after the the change to our own library with fewer allocs, needs benchmarking)

Reply:
You can't use `sdk.ValAddress]` as a `map` key though. It has to be string or something else compare-able.